packages/data-stores/src/woocommerce-eligibility/selectors.ts

```diff
@@ -0,0 +1,102 @@
+import { select } from '@wordpress/data';
+import { get } from 'lodash';
+import { STORE_KEY, statusMapping } from './constants';
+import { TransferEligibilityWarning, TransferEligibilityError, TransferEligibility } from './types';
+import type { State } from './reducer';
+
+export const getState = ( state: State ) => state;
+
+export const getAutomatedTransferEligibility = (
+	state: State,
+	siteId: number | null
+): TransferEligibility | null => {
+	if ( ! state || ! state.transferEligibility ) {
+		return null;
+	}
+
+	if ( ! siteId ) {
+		return null;
+	}
+	return state.transferEligibility[ siteId ];
+};
+
+export const getEligibilityHolds = (
+	state: State,
+	siteId: number | null
+): TransferEligibilityError[] | null => {
+	const transferEligibility = select( STORE_KEY ).getAutomatedTransferEligibility( siteId );
+
+	if ( ! transferEligibility ) {
+		return null;
+	}
+
+	const { errors } = transferEligibility;
+
+	if ( ! errors ) {
+		return null;
+	}
+
+	errors.map( ( { code }: { code: string } ) => {
+		return get( statusMapping, code, '' );
+	} );
+
+	return errors;
+};
+
+export const getEligibilityWarnings = (
+	state: State,
+	siteId: number | null
+): TransferEligibilityWarning[] | null => {
+	const transferEligibility = select( STORE_KEY ).getAutomatedTransferEligibility( siteId );
+
+	if ( ! transferEligibility ) {
+		return null;
+	}
+
+	const { warnings } = transferEligibility;
+
+	if ( ! warnings ) {
+		return null;
+	}
+
+	// combine plugin and theme warnings into one list
+	let combined: TransferEligibilityWarning[] = [];
+	for ( const key of Object.keys( warnings ) ) {
+		combined = combined.concat( warnings[ key ] );
+	}
+
+	return combined.map( ( { description, name, supportUrl, id } ) => ( {
+		id,
+		name,
+		description,
+		supportUrl,
+	} ) );
+};
+
+export const getNonSubdomainWarnings = (
+	state: State,
+	siteId: number | null
+): TransferEligibilityWarning[] | null => {
+	const eligibilityWarnings = select( STORE_KEY ).getEligibilityWarnings( siteId );
+
+	if ( ! eligibilityWarnings ) {
+		return null;
+	}
+
+	return eligibilityWarnings?.filter( ( { id } ) => id !== 'wordpress_subdomain' ) ?? null;
```

Comment:
The `?.` and `??` operators seem to be redundant because an `undefined` or `null` value has already been handled few lines above.